src/Type.php

```diff
@@ -469,8 +469,10 @@ public function save(EntityInterface $entity, $options = [])
         $type = $this->connection()->getIndex()->getType($this->name());
         $id = $entity->id ?: null;
 
-        $data = $entity->toArray();
-        unset($data[$id]);
+        $keys = array_keys($entity->toArray());
+        $metaKeys = ['id', '_version'];
+        $data = $entity->extract(array_diff($keys, $metaKeys));
```

Comment:
Ok what is the best approach from here? Shall I revert this commit (locally), make a new one and then create another pull request for that new commit?